css/src/components/layout.scss

```diff
@@ -207,3 +220,55 @@ $three-quarters-widescreen: math.div($breakpoint-widescreen, 4) * 3;
 		}
 	}
 }
+
+@mixin constrained-layout-child {
+	position: sticky;
+	inset-block-start: 0;
+	overflow-x: hidden;
+	overflow-y: auto;
+	-webkit-overflow-scrolling: touch;
+}
+
+@include tablet {
+	.layout.layout-constrained {
+		&.layout-twin,
+		&.layout-sidecar-left,
+		&.layout-sidecar-right {
+			// 👇 minus two pixel at the end to account for percentage points and rounding, one px generally suffices though
+			--atlas-contained-height: calc(
+				var(--window-inner-height) - var(--atlas-header-height) - var(--atlas-footer-height) - 2px
+			);
+		}
+
+		&.layout-twin,
+		&.layout-sidecar-right {
+			.layout-body-main,
+			.layout-body-aside {
+				@include constrained-layout-child;
+			}
+		}
+
+		&.layout-sidecar-left {
+			.layout-body-menu,
+			.layout-body-main {
+				@include constrained-layout-child;
+			}
+		}
+	}
+}
+
+// Because the holy grail has two rows (containing menu main, menu aside) on tablet, we cannot apply height constrains at that size
```

Comment:
```suggestion
// Because the holy grail has two rows (containing menu main, menu aside) on tablet, we cannot apply height constraints at that size
```